scripts/start.js

```diff
@@ -66,62 +69,90 @@ function clearConsole() {
 }
 
 var compiler = webpack(config, handleCompile);
-compiler.plugin('invalid', function () {
-  clearConsole();
-  console.log('Compiling...');
-});
-compiler.plugin('done', function (stats) {
-  clearConsole();
-  var hasErrors = stats.hasErrors();
-  var hasWarnings = stats.hasWarnings();
-  if (!hasErrors && !hasWarnings) {
-    console.log(chalk.green('Compiled successfully!'));
-    console.log();
-    console.log('The app is running at http://localhost:3000/');
-    console.log();
-    return;
-  }
+detect(PORT, (error, _port) => {
 
-  var json = stats.toJson();
-  var formattedErrors = json.errors.map(message =>
-    'Error in ' + formatMessage(message)
-  );
-  var formattedWarnings = json.warnings.map(message =>
-    'Warning in ' + formatMessage(message)
-  );
+  if (PORT !== _port) {
+    var rl = readline.createInterface({
+      input: process.stdin,
+      output: process.stdout
+    });
 
-  if (hasErrors) {
-    console.log(chalk.red('Failed to compile.'));
-    console.log();
-    if (formattedErrors.some(isLikelyASyntaxError)) {
-      // If there are any syntax errors, show just them.
-      // This prevents a confusing ESLint parsing error
-      // preceding a much more useful Babel syntax error.
-      formattedErrors = formattedErrors.filter(isLikelyASyntaxError);
-    }
-    formattedErrors.forEach(message => {
-      console.log(message);
-      console.log();
+    rl.question('Something is already running at http://localhost:' + PORT + '. Would you like to run the app at another port instead? [Y/n] ', (answer) => {
+      if(answer === 'Y') {
```

Comment:
`Y` being capital means it’s the default answer. So specifying nothing, or specifying `y`, `yes`, `Y`, or `YES` should all be treated as `y`. Specifying `n`, `N`, `no`, or `NO` should all be treated as `no`.

You can see an example of this [here](https://github.com/facebookincubator/create-react-app/blob/662fdb7055fa15c8461d1b3605b2e2c8d590d1e9/scripts/eject.js#L28-L31). The only difference is that there, `no` is the default, so we look for explicit `yes`. Here, it’s the other way around.